lua/lspconfig/haxe_language_server.lua

```diff
@@ -0,0 +1,61 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+configs.haxe_language_server = {
+  default_config = {
+    cmd = {"haxe-language-server"};
+    filetypes = {"haxe"};
+    root_dir = util.root_pattern("build.hxml");
+    settings = {
```

Comment:
nit but does

```lua
settings = {
    haxe = {
        executable = "haxe"; 
    };
}
```

work? More consistent with other config styles. Also is this setting required?

Reply:
Good call on the style consistency. After some experimentation, I noticed that the setting is in fact _not_ required, so I'll just remove it to keep things simple.

Reply:
Spoke too soon—the setting is required after all. I will, however, update the style here.